internal/playback/simulation.go

```diff
@@ -0,0 +1,42 @@
+package playback
+
+import (
+	"log"
+	"time"
+)
+
+const defaultRate = 10 * time.Millisecond
+
+func SimulatePlayback(changer chan time.Duration, interruptC chan struct{}) (stoppedAt <-chan time.Duration, finished <-chan struct{}) {
+	stoppedAtNotifier := make(chan time.Duration)
+	finishedNotifier := make(chan struct{})
+
+	go playback(changer, interruptC, stoppedAtNotifier, finishedNotifier)
+
+	return stoppedAtNotifier, finishedNotifier
+}
+
+func playback(changerC chan time.Duration, interruptC chan struct{}, stoppedAt chan time.Duration, finished chan struct{}) {
+	ticker := time.NewTicker(defaultRate)
```

Comment:
I like this approach using `time.Ticker` but it's too many channels and synchronization. I recommend embedding playback functionality into the playlist.

Reply:
Best: for select in single goroutine
Use command pattern via select + use it for song finish
What happens with ticker if other job takes too long
see Time after
see context deadline (+context timeout)

Reply:
Make service structure